psycop/common/feature_generation/sequences/timeseries_windower_python/events/temporal_event.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import datetime as dt
+
+    from psycop.common.feature_generation.sequences.timeseries_windower_python.patient import (
+        Patient,
+    )
+
+
+@dataclass
+class TemporalEvent:
+    timestamp: dt.datetime
+    source: str  # E.g. "lab"/"diagnosis"
+    name: str | None  # E.g. "Hba1c"/"hypertension". Is optional, since e.g. diagnoses don't have a name, only a source and value.
+    value: float | str  # 1/0 for booleans, numeric value for numeric events
```

Comment:
I would use True/False for booleans. Any reason to not do that?

Reply:
In principle, we could imagine subclasses along the lines of DiagnosisEvent, where you specify the type more explicitly.